GitUI/UserControls/RevisionGrid/CopyContextMenuItem.cs

```diff
@@ -64,29 +70,38 @@ void OnDropDownOpening(object sender, EventArgs e)
 
                     foreach (var name in tagNames)
                     {
-                        AddItem(name, _ => name, Images.Tag);
+                        AddImmutableItem(name, Images.Tag);
                     }
 
                     DropDownItems.Add(new ToolStripSeparator());
                 }
 
                 // Add other items
-                AddItem($"{Strings.CommitHash}     ({revision.ObjectId.ToShortString()}...)", r => r.Guid, Images.CommitId, Keys.Control | Keys.C);
-                AddItem($"{Strings.Message}     ({revision.Subject.ShortenTo(30)})", r => r.Body ?? r.Subject, Images.Message);
-                AddItem($"{Strings.Author}     ({revision.Author})", r => r.Author, Images.Author);
+                AddItem(Strings.CommitHash,  (GitRevision r) => r.Guid, Images.CommitId, Keys.Control | Keys.C);
+                AddItem(Strings.Message,     r => r.Body ?? r.Subject, Images.Message);
+                AddItem(Strings.Author,      r => r.Author, Images.Author);
+                AddItem(Strings.AuthorDate,  r => r.AuthorDate.ToString(), Images.Date);
+                AddItem(Strings.CommitDate,  r => r.CommitDate.ToString(), Images.Date);
+            }
 
-                if (revision.AuthorDate == revision.CommitDate)
+            void AddImmutableItem(string displayText, Image image, Keys shortcutKeys = Keys.None, bool showShortcutKeys = true)
+            {
+                var item = new ToolStripMenuItem
                 {
-                    AddItem($"{Strings.Date}     ({revision.CommitDate})", r => r.CommitDate.ToString(), Images.Date);
-                }
-                else
+                    Text = displayText,
+                    ShortcutKeys = shortcutKeys,
+                    ShowShortcutKeys = showShortcutKeys,
+                    Image = image
+                };
+                item.Click += delegate
                 {
-                    AddItem($"{Strings.AuthorDate}     ({revision.AuthorDate})", r => r.AuthorDate.ToString(), Images.Date);
-                    AddItem($"{Strings.CommitDate}     ({revision.CommitDate})", r => r.CommitDate.ToString(), Images.Date);
-                }
+                    Clipboard.SetText(displayText);
+                };
+
+                DropDownItems.Add(item);
             }
 
-            void AddItem(string displayText, Func<GitRevision, string> clipboardText, Image image, Keys shortcutKeys = Keys.None, bool showShortcutKeys = true)
+            void AddItem(string displayText, Func<GitRevision, string> extractTextFunc, Image image, Keys shortcutKeys = Keys.None, bool showShortcutKeys = true)
```

Comment:
`AddImmutableItem` and `AddItem` methods look very similar, with only difference in the click handler. Please combine them.